src/Microsoft.Extensions.ProjectModel.Sources/MsBuild/MsBuildProjectContext.cs

```diff
@@ -16,23 +17,29 @@ internal class MsBuildProjectContext : IProjectContext
         private const string EmbedItemName = "EmbeddedResource";
         private const string FullPathMetadataName = "FullPath";
 
-        private readonly ProjectInstance _project;
-        private readonly string _name;
+        private Lazy<MsBuildProjectDependencyProvider> _dependencyProvider;
+        private IEnumerable<DependencyDescription> _packageDependencies;
+        private IEnumerable<ResolvedReference> _compilationAssemblies;
+
+        protected ProjectInstance Project { get; }
+        protected string Name { get; }
 
         public MsBuildProjectContext(string name, string configuration, ProjectInstance project)
         {
-            _project = project;
+            Project = project;
 
             Configuration = configuration;
-            _name = name;
+            Name = name;
+            _dependencyProvider = new Lazy<MsBuildProjectDependencyProvider>(() =>
```

Comment:
Constructing `MsBuildProjectDependencyProvider` is incredibly cheap unless there's a bit in the implementation that I'm missing. Probably more expensive here to create a Lazy than the actual class 😉